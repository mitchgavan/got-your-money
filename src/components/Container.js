import styled from 'styled-components'
import { Box } from 'grid-styled'

export default styled(Box)`
  max-width: 1024px;
  margin-left: auto;
  margin-right: auto;
`
