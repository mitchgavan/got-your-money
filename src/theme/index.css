body {
  font-family: 'Roboto', sans-serif;
  background-color: #f3f4f5;
}
