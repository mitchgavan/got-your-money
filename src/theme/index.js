export default {
  breakpoints: [
    32, 48, 64
  ],
  space: [
    0, 6, 12, 18, 24
  ],
  fontSizes: [
    12, 16, 18, 24, 36, 72
  ],
  colors: {
    black: '#111',
    blue: '#07c',
    green: '#3adb76'
  }
}
