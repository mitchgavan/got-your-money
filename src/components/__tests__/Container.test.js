import React from 'react'
import Container from '../Container'
import { mount } from 'enzyme'
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import theme from '../../theme'

let component

beforeEach(() => {
  component = mount(
    <Container
      theme={theme}
      flex="1 1 auto"
      p={2}
      mb={2}
      width={[1 / 2, 1 / 3, 1 / 4]}
    />
  )
})

it('renders correctly', () => {
  expect(toJson(component)).toMatchSnapshot()
})

test('it applies styles according to passed props', () => {
  expect(component).toHaveStyleRule('flex', '1 1 auto')
  expect(component).toHaveStyleRule('padding', theme.space[2] + 'px')
  expect(component).toHaveStyleRule('margin-bottom', theme.space[2] + 'px')
  expect(component).toHaveStyleRule('width', '50%')
})
