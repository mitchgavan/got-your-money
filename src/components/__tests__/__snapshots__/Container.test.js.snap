// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  box-sizing: border-box;
  width: 50%;
  margin-bottom: 12px;
  padding: 12px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c0 {
  max-width: 1024px;
  margin-left: auto;
  margin-right: auto;
}

@media screen and (min-width:32px) {
  .c1 {
    width: 33.33333333333333%;
  }
}

@media screen and (min-width:48px) {
  .c1 {
    width: 25%;
  }
}

<Styled(Box)
  flex="1 1 auto"
  mb={2}
  p={2}
  theme={
    Object {
      "breakpoints": Array [
        32,
        48,
        64,
      ],
      "colors": Object {
        "black": "#111",
        "blue": "#1779ba",
        "green": "#3adb76",
        "lightestGrey": "#f3f4f5",
        "white": "#fefefe",
      },
      "fontSizes": Array [
        10,
        12,
        16,
        18,
        24,
        36,
        72,
      ],
      "space": Array [
        0,
        6,
        12,
        18,
        24,
        32,
      ],
    }
  }
  width={
    Array [
      0.5,
      0.3333333333333333,
      0.25,
    ]
  }
>
  <Box
    className="c0"
    flex="1 1 auto"
    mb={2}
    p={2}
    theme={
      Object {
        "breakpoints": Array [
          32,
          48,
          64,
        ],
        "colors": Object {
          "black": "#111",
          "blue": "#1779ba",
          "green": "#3adb76",
          "lightestGrey": "#f3f4f5",
          "white": "#fefefe",
        },
        "fontSizes": Array [
          10,
          12,
          16,
          18,
          24,
          36,
          72,
        ],
        "space": Array [
          0,
          6,
          12,
          18,
          24,
          32,
        ],
      }
    }
    width={
      Array [
        0.5,
        0.3333333333333333,
        0.25,
      ]
    }
  >
    <div
      blacklist={
        Array [
          "width",
          "w",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "fontSize",
          "f",
          "color",
          "bg",
          "flex",
          "order",
          "alignSelf",
          "css",
        ]
      }
      className="c0 c1"
      flex="1 1 auto"
      mb={2}
      p={2}
      theme={
        Object {
          "breakpoints": Array [
            32,
            48,
            64,
          ],
          "colors": Object {
            "black": "#111",
            "blue": "#1779ba",
            "green": "#3adb76",
            "lightestGrey": "#f3f4f5",
            "white": "#fefefe",
          },
          "fontSizes": Array [
            10,
            12,
            16,
            18,
            24,
            36,
            72,
          ],
          "space": Array [
            0,
            6,
            12,
            18,
            24,
            32,
          ],
        }
      }
      width={
        Array [
          0.5,
          0.3333333333333333,
          0.25,
        ]
      }
    >
      <Clean.div
        blacklist={
          Array [
            "width",
            "w",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "fontSize",
            "f",
            "color",
            "bg",
            "flex",
            "order",
            "alignSelf",
            "css",
          ]
        }
        className="c0 c1"
        flex="1 1 auto"
        is="div"
        mb={2}
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              32,
              48,
              64,
            ],
            "colors": Object {
              "black": "#111",
              "blue": "#1779ba",
              "green": "#3adb76",
              "lightestGrey": "#f3f4f5",
              "white": "#fefefe",
            },
            "fontSizes": Array [
              10,
              12,
              16,
              18,
              24,
              36,
              72,
            ],
            "space": Array [
              0,
              6,
              12,
              18,
              24,
              32,
            ],
          }
        }
        width={
          Array [
            0.5,
            0.3333333333333333,
            0.25,
          ]
        }
      >
        <div
          className="c0 c1"
        />
      </Clean.div>
    </div>
  </Box>
</Styled(Box)>
`;
